lib/spack/spack/cmd/create.py

```diff
@@ -328,14 +328,28 @@ def configure_args(self):
         args = []
         return args"""
 
-    def __init__(self, name, *args, **kwargs):
+    def __init__(self, name, url, *args, **kwargs):
         # If the user provided `--name r-rcpp`, don't rename it r-r-rcpp
         if not name.startswith('r-'):
             # Make it more obvious that we are renaming the package
             tty.msg("Changing package name from {0} to r-{0}".format(name))
             name = 'r-{0}'.format(name)
 
-        super(RPackageTemplate, self).__init__(name, *args, **kwargs)
+            r_name = parse_name(url)
+
+            if re.search(r'.*\.r-project\.org/src/contrib', url):
+                url = re.sub(r'cran\.r-project', 'cloud.r-project', url)
+                list_url = "https://cloud.r-project.org/src/contrib/Archive/{0}".format(r_name)
+                self.url_line = '    url      = "{0}"\n'\
+                    '    list_url = "{1}"'.format(url, list_url)
+            elif re.search(r'.*bioconductor.org', url):
+                git = "https://git.bioconductor.org/packages/{0}".format(r_name)
```

Comment:
IMO It is not appropriate or desirable to hard-code URL paths into core Spack commands.  

If these are necessary for R packages then I suggest they be added to `RPackage` and accessed here.